alf/utils/common.py

```diff
@@ -856,3 +856,26 @@ def flattened_size(spec):
     """
     # np.prod(()) == 1.0, need to convert to np.int64
     return np.int64(np.prod(spec.shape))
+
+
+def is_inside_docker_container():
+    """Return whether the current process is running inside a docker container.
+
+    See discussions at `<https://stackoverflow.com/questions/23513045/how-to-check-if-a-process-is-running-inside-docker-container>`_
+    """
+    return os.path.exists("/.dockerenv")
+
+
+def check_numerics(nested):
```

Comment:
There are several places in the current code that have a simpler check_numerics. Maybe replace them with this function?

Reply:
This is good for nested one. The other places are not checking nested.